libraries/bdp-access-proxy/bdp-access-impl/src/main/java/com/bosch/osmi/bdp/access/impl/model/ComponentImpl.java

```diff
@@ -0,0 +1,174 @@
+import com.blackducksoftware.sdk.protex.project.bom.BomComponent;
+import com.blackducksoftware.sdk.protex.project.bom.BomComponentType;
+import com.bosch.osmi.bdp.access.api.model.Component;
+import com.bosch.osmi.bdp.access.api.model.License;
+import com.bosch.osmi.bdp.access.impl.BdpApiAccessImpl;
+
+import java.text.SimpleDateFormat;
+import java.util.Date;
+import java.util.List;
+
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+
+/**
+ * @author muj1be
+ * @since 11/18/15.
+ */
+public class ComponentImpl implements Component {
+    private static final Logger LOGGER = LogManager.getLogger(ComponentImpl.class);
+    private static final String UNSPECIFIED = "Unspecified";
+    private static final String EMPTY = "";
+
+    private final BomComponent source;
+    private final String projectId;
+    private final BdpApiAccessImpl access;
+
+    public ComponentImpl(BomComponent source, String projectId, BdpApiAccessImpl access) {
+        this.source = source;
+        this.projectId = projectId;
+        this.access = access;
+    }
+
+    @Override
+    public String getName() {
+        return source.getComponentName();
+    }
+
+    @Override
+    public License getLicense() {
+        License result = new LicenseImpl(source.getLicenseInfo(), access);
+        return result;
+    }
+
+    @Override
+    public String getComponentVersion() {
+        if(source.getBomVersionName() == null){
+            return EMPTY;
+        }else if (source.getBomVersionName().equals(UNSPECIFIED)) {
+            return EMPTY;
+        } else {
+            return source.getBomVersionName();
+        }
+    }
+
+    @Override
+    public String getComponentHomePage() {
+        try {
+            String homepage = EMPTY;
+            if (source.getType().equals(BomComponentType.PUBLIC)) {
+                ComponentKey componentKey = source.getComponentKey();
+                homepage = access.getComponentApi().getComponentByKey(componentKey).getHomePage();
+                if(homepage == null) {
+                    homepage = EMPTY;
+                }
+            }
+            return homepage;
+        } catch (SdkFault sdkFault) {
+            LOGGER.error("Unable to retrieve project data. Reason \n" + sdkFault.getMessage());
+            return EMPTY;
+        }
+    }
+
+    @Override
+    public String getComponentComment() {
+        try {
+            ComponentKey key = source.getComponentKey();
+            String result = access.getBomApi().getComponentComment(projectId, key);
+            if(result == null){
+                result = EMPTY;
+            }
+            return result;
+        } catch (SdkFault sdkFault) {
+            LOGGER.error("Unable to retrieve project data. Reason \n" + sdkFault.getMessage());
+            return EMPTY;
+        }
+    }
+
+    @Override
+    public String getUsageLevel() {
+        String usageLevel = "";
+        //one component is expected to have only one usage
```

Comment:
Yes you are right. I have adapted the code to make the assumptions behind more explicit